### PR TITLE
Filter deleted files and avoid merge commits

### PR DESCRIPTION
**Pull request by parlir**

### CODE REVIEW

File: src/compile.rs [hunk 1]
```diff
@@ -11,7 +11,11 @@ pub fn script_gen(location_command: &str, hook: &str, hook_config_file: &str) ->
 #!/bin/sh
 # {}
 # Hook created by precommit
-{} run {} {}
+if  git rev-parse -q --verify MERGE_HEAD; then
```
Comment (idubrov) [inline src/compile.rs]: Can you do that from inside the tool itself, so you have more control (and don't change hook script, which is nice from backwards compatibility point of view)?

Comment (parlir) [reply src/compile.rs]: Yea I could add arg parameter for this

Comment (idubrov) [reply src/compile.rs]: No, I mean, can you get that info in the tool itself, without `get rev-pase`?

Comment (parlir) [reply src/compile.rs]: I probably can.

**Pull request closed without merging**